### PR TITLE
ci: deflake make slurmcluster, hopefully

### PR DESCRIPTION
**Pull request by rb-determined-ai**
The `make slurmcluster` step in CI has two fairly common failure modes:
  - The `gcloud compute scp` command to fetch the launcher token fails [saying the remote file does not exist](https://app.circleci.com/pipelines/github/determined-ai/determined/59943/workflows/a719335a-9783-4988-94c1-48316a227613/jobs/2894155).  That feels like it might be some sort of startup problem rather than the file only existing in some VMs.
  - The `gcloud compute ssh` command to start launcher.service sometimes [fails complaining about a timeout](https://app.circleci.com/pipelines/github/determined-ai/determined/60237/workflows/a2bc119e-bacd-4d54-a6ea-b14c2bc5e0fe/jobs/2916385).  Additional retries may solve this.

I'm hoping to fix the flakes by introducing a small retry loop around these commands.  Unfortunately, I can't be sure this is going to solve the problems reliably.  Time will tell.

### CODE REVIEW

File: tools/slurm/scripts/slurmcluster.sh [hunk 1]
```diff
@@ -15,6 +15,29 @@ export OPT_LAUNCHER_PORT=8081
...
+tryloop() {
+    while [ "$1" != "--" ]; do
+        tryloop_attempt "$@" && return
+        echo "retrying in $1..." >&2
+        sleep "$1"
+        shift
+    done
+    # Final attempt.
+    tryloop_attempt "$@"
+}
```
Comment (maxrussell) [inline tools/slurm/scripts/slurmcluster.sh:39]: This is sick. Love this as an explicit exponential backoff config declaration.

File: tools/slurm/scripts/slurmcluster.sh [hunk 2]
```diff
@@ -146,11 +169,11 @@ echo "Started bidirectional tunnels to $INSTANCE_NAME"
 # Grab launcher token.
 REMOTE_TOKEN_SOURCE=/opt/launcher/jetty/base/etc/.launcher.token
 LOCAL_TOKEN_DEST=$TEMPDIR/.launcher.token
-gcloud compute scp --quiet --zone "us-west1-b" --project "determined-ai" root@$INSTANCE_NAME:$REMOTE_TOKEN_SOURCE $LOCAL_TOKEN_DEST
+tryloop 1 10 30 -- gcloud compute scp --quiet --zone "us-west1-b" --project "determined-ai" root@$INSTANCE_NAME:$REMOTE_TOKEN_SOURCE $LOCAL_TOKEN_DEST
 echo "Copied launcher token to $LOCAL_TOKEN_DEST"
 # The launcher service verifies communication with the launcher, so just have to be sure it is started.
 # Also show the status for extra confirmaiton in the logs of the state.
-gcloud compute ssh --zone "$ZONE" "$INSTANCE_NAME" --project "$PROJECT" -- "sudo systemctl start launcher.service ; systemctl status launcher.service --no-pager"
+tryloop 1 10 30 -- gcloud compute ssh --zone "$ZONE" "$INSTANCE_NAME" --project "$PROJECT" -- "sudo systemctl start launcher.service ; systemctl status launcher.service --no-pager"
```
Comment (maxrussell) [inline tools/slurm/scripts/slurmcluster.sh:176]: Maybe we want to try 1 a few times before waiting so long. I'm cool either way, though.

**Review by maxrussell — APPROVED** [review]
LGTM.

File: tools/slurm/scripts/slurmcluster.sh (hunk 2 above)
Comment (rb-determined-ai) [reply tools/slurm/scripts/slurmcluster.sh:176]: I can do that

File: tools/slurm/scripts/slurmcluster.sh (hunk 1 above)
Comment (rb-determined-ai) [reply tools/slurm/scripts/slurmcluster.sh:39]: thanks!

**Pull request merged**